src/ssh/cockpitsshrelay.c

```diff
@@ -502,10 +482,10 @@ prompt_for_host_key (CockpitSshData *data)
                              host, port);
   prompt = g_strdup_printf ("MD5 Fingerprint (%s):", data->host_key_type);
 
-  reply = prompt_with_authorize (data, prompt, message, data->host_fingerprint, TRUE);
+  reply = prompt_with_authorize (data, prompt, message, data->host_fingerprint, data->host_key, TRUE);
```

Comment:
This will change the behavior. Right now if a host key is passed from the json. That host key will always be tried even if the host key already exists in the known hosts file. With a special value for "" to force a host key mismatch and the connection to fail. 

If I'm reading this right, this will only prompt if the host-key isn't found, that will change the current behavior. Generally i wouldn't expect that to be too much of a problem in practice. Except I think some versions of cockpit might rely setting host-key: ""  to force fail even when a host key might have otherwise been found.